packages/licenses/src/getPkgInfo.ts

```diff
@@ -0,0 +1,356 @@
+  if (Array.isArray(field)) {
+    const licenses = field
+    const licenseTypes = licenses.reduce((listOfLicenseTypes, license) => {
+      const type = coerceToString(license.type) ?? coerceToString(license.name)
+      if (type) {
+        listOfLicenseTypes.push(type)
+      }
+      return listOfLicenseTypes
+    }, [])
+
+    if (licenseTypes.length > 1) {
+      const combinedLicenseTypes = licenseTypes.join(' OR ') as string
+      return `(${combinedLicenseTypes})`
+    }
+
+    return licenseTypes[0] ?? null
+  } else {
+    return (field as { type: string })?.type ?? coerceToString(field)
+  }
+}
+
+/**
+ * Reads the license field or LICENSE file from
+ * the directory of the given package manifest
+ *
+ * If the package.json file is missing the `license`-property
+ * the root of the manifest directory will be scanned for
+ * files named listed in the array LICENSE_FILES and the
+ * contents will be returned.
+ *
+ * @param {*} pkg the package to check
+ * @returns Promise<LicenseInfo>
+ */
+async function parseLicense (
+  pkg: {
+    manifest: PackageManifest
+    files:
+    | { local: true, files: Record<string, string> }
+    | { local: false, files: Record<string, PackageFileInfo> }
+  },
+  opts: { cafsDir: string }
+): Promise<LicenseInfo> {
+  let licenseField: unknown = pkg.manifest.license
+  if ('licenses' in pkg.manifest) {
+    licenseField = (
+      pkg.manifest as PackageManifest & {
+        licenses: unknown
+      }
+    ).licenses
+  }
+  const license = parseLicenseManifestField(licenseField)
+
+  // check if we discovered a license, if not attempt to parse the LICENSE file
+  if (!license || /see license/i.test(license)) {
+    for (const filename of LICENSE_FILES) {
+      try {
+        let licenseContents: Buffer | undefined
+        if (pkg.files.local) {
+          const licensePackageFileInfo = pkg.files.files[filename]
+          licenseContents = await readFile(licensePackageFileInfo)
+        } else {
+          const licensePackageFileInfo = pkg.files.files[filename]
+          licenseContents = await readLicenseFileFromCafs(
+            opts.cafsDir,
+            licensePackageFileInfo.integrity
+          )
+        }
+
+        return {
+          name: 'Unknown',
+          licenseFile: licenseContents?.toString('utf-8'),
+        }
+      } catch (err) {
+        // Silently ignore the error when failed
+        // to read the contents of a license file
+      }
+    }
+  }
+
+  return { name: license ?? 'Unknown' }
+}
+
+/**
+ * Fetch a file by integrity id from the content-addressable store
+ * @param file the file to fetch
+ * @param opts the options
+ * @returns Promise<Buffer>
+ */
+async function readLicenseFileFromCafs (cafsDir: string, fileIntegrity: string) {
+  const fileName = getFilePathByModeInCafs(cafsDir, fileIntegrity, 0)
+
+  try {
+    const fileContents = await readFile(fileName)
+    return fileContents
+  } catch (err: any) { // eslint-disable-line
```

Comment:
I don't think this try/catch is needed at all. The only scenario when this may happen is when someone manually removes the license from the content-addressable store, which should never happen.

Reply:
Simplified things